eventlet/support/greendns.py

```diff
@@ -178,16 +187,16 @@ def _readlines(self):
 
         Note that this performs disk I/O so can be blocking.
         """
-        lines = []
         try:
-            with open(self.fname, 'rU') as fp:
-                for line in fp:
-                    line = line.strip()
-                    if line and line[0] != '#':
-                        lines.append(line)
+            with open(self.fname, 'rb') as fp:
+                fdata = fp.read()
         except (IOError, OSError):
-            pass
-        return lines
+            return []
+
+        if not isinstance(fdata, str):  # Python 3
```

Comment:
> Yes, and on python 2, isinstance(u'text', str) is False

How come this check is relevant? Please explain chain of thought.

_readlines does not have to return exact `str` type. It must return conceptually text, which is `unicode` in 2 and `str` in 3.

> There should really be some function (e.g. bytes_to_str)

`.decode()` is that function. It works for both Python versions types representing binary.